test/host/wasi_nn/wasi_nn.cpp

```diff
@@ -0,0 +1,323 @@
+#include "common/defines.h"
+#include <gtest/gtest.h>
+
+#include "host/wasi_nn/wasinnenv.h"
+#include "host/wasi_nn/wasinnfunc.h"
+#include <algorithm>
+#include <cstdint>
+#include <fstream>
+#include <numeric>
+#include <vector>
+
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+using WasmEdge::Host::WASINN::ErrNo;
+namespace {
+
+inline std::vector<uint8_t> readBinariesFromDisk(const std::string file_path) {
+  std::ifstream instream(file_path, std::ios::in | std::ios::binary);
+  std::vector<uint8_t> data((std::istreambuf_iterator<char>(instream)),
+                            std::istreambuf_iterator<char>());
+  return data;
+}
+
+template <typename T>
+void writeBinaries(WasmEdge::Runtime::Instance::MemoryInstance &MemInst,
+                   std::vector<T> Binaries, uint32_t Ptr) noexcept {
+  std::copy(Binaries.begin(), Binaries.end(), MemInst.getPointer<T *>(Ptr));
+}
+
+void writeUInt32(WasmEdge::Runtime::Instance::MemoryInstance &MemInst,
+                 uint32_t value, uint32_t &Ptr) {
+  uint32_t *BufPtr = MemInst.getPointer<uint32_t *>(Ptr);
+  *BufPtr = value;
+  Ptr += 4;
+}
+
+void writeFatPointer(WasmEdge::Runtime::Instance::MemoryInstance &MemInst,
+                     uint32_t PtrVal, uint32_t PtrSize, uint32_t &Ptr) {
+  writeUInt32(MemInst, PtrVal, Ptr);
+  writeUInt32(MemInst, PtrSize, Ptr);
+}
+
+template <typename T>
+std::vector<size_t> classSort(const std::vector<T> &array) {
+  std::vector<size_t> indices(array.size());
+  std::iota(indices.begin(), indices.end(), 0);
+  std::sort(indices.begin(), indices.end(),
+            [&array](int left, int right) -> bool {
+              // sort indices according to corresponding array element
+              return array[left] >= array[right];
+            });
+
+  return indices;
+}
+} // namespace
+
+TEST(WasiNNTest, OpenVINOBackend) {
+  WasmEdge::Host::WASINN::WasiNNEnvironment Env;
+  WasmEdge::Runtime::Instance::MemoryInstance MemInst(
+      WasmEdge::AST::MemoryType(400));
+  std::array<WasmEdge::ValVariant, 1> Errno = {UINT32_C(0)};
+
+  std::vector<uint8_t> XmlRead, WeightRead, TensorData;
+  std::vector<uint32_t> TensorDim{1, 3, 224, 224};
+  uint32_t BuilderPtr = UINT32_C(0);
+  uint32_t LoadEntryPtr;
+  uint32_t SetInputEntryPtr;
+  uint32_t StorePtr = UINT32_C(65536);
+
+  LoadEntryPtr = BuilderPtr;
+  XmlRead = readBinariesFromDisk("./wasinn_openvino_fixtures/mobilenet.xml");
+  WeightRead = readBinariesFromDisk("./wasinn_openvino_fixtures/mobilenet.bin");
+
+  writeFatPointer(MemInst, StorePtr, XmlRead.size(), BuilderPtr);
+  writeFatPointer(MemInst, StorePtr + XmlRead.size(), WeightRead.size(),
+                  BuilderPtr);
+
+  // wasi-nn load test
+  WasmEdge::Host::WasiNNLoad WasiNNLoad(Env);
+  // meaningless binaries
+  {
+    EXPECT_TRUE(WasiNNLoad.run(
+        &MemInst,
+        std::initializer_list<WasmEdge::ValVariant>{
+            LoadEntryPtr, UINT32_C(2), UINT32_C(0), UINT32_C(0), BuilderPtr},
```

Comment:
I think the last argument is the returned graph ID pointer?

Reply:
Yes. The var naming is a little bit confusing here...
`BuilderPtr` is an increasing pointer from `0x0000`.  It points to a memory to place the returned graph ID here.